core-java/src/main/java/org/spine3/server/aggregate/Aggregate.java

```diff
@@ -274,18 +297,56 @@ public void restore(SnapshotOrBuilder snapshot) {
         setState(stateToRestore, snapshot.getVersion(), snapshot.getWhenModified());
     }
 
+    private void putUncommitted(EventRecord record) {
+        uncommittedEvents.add(record);
+    }
+
     /**
-     * @return immutable view of records for uncommitted events
+     * Returns all uncommitted events (including state-neutral).
+     *
+     * @return immutable view of records for all uncommitted events
+     * @see #getStateNeutralEventClasses()
      */
     @CheckReturnValue
-    public List<EventRecord> getUncommittedEvents() {
+    public List<EventRecord> getAllUncommittedEvents() {
         return ImmutableList.copyOf(uncommittedEvents);
     }
 
     /**
-     * Returns and clears the events that were uncommitted before the call of this method.
+     * Returns uncommitted events (excluding state-neutral).
+     *
+     * @return immutable view of records for applicable uncommitted events
+     * @see #getStateNeutralEventClasses()
+     */
+    public Collection<EventRecord> getApplicableUncommittedEvents() {
```

Comment:
Why do we want this method being `public`? What's the interest of having only state-changing events _outside_ the aggregate?